src/test.rs

```diff
@@ -9,24 +9,41 @@ use pact::types::{Numeric, PactType, StringLike};
 use std::string::{String, ToString};
 use std::vec::Vec;
 
-#[test]
-fn it_works_encode() {
-    let method = Method {
-        name: "method_test".to_string(),
-        block_cooldown: None,
-        constraints: None,
-    };
-    let mut methods: Vec<(String, Method)> = Default::default();
-    methods.push((method.name.clone(), method.clone()));
+fn init_method(name: &str, block_cooldown: Option<u32>, constraints: Option<Vec<u8>>) -> Method {
+    Method {
+        name: name.to_string(),
+        block_cooldown,
+        constraints,
+    }
+}
 
-    let module = Module {
-        name: "module_test".to_string(),
-        block_cooldown: None,
+fn init_module(name: &str, block_cooldown: Option<u32>, methods: Vec<(String, Method)>) -> Module {
```

Comment:
Similar to `Method` above, could this be implemented as a constructor for `Module`

Reply:
Good idea, will update about this

Reply:
+1. Have a look at this pattern also: https://doc.rust-lang.org/1.0.0/style/ownership/builders.html